esmvaltool/cmorizers/data/formatters/datasets/esacci_biomass.py

```diff
@@ -0,0 +1,174 @@
+        # save each year to a separate output file
+        timecoord = cube.coord("time")
+        for time in timecoord.units.num2date(timecoord.points):
+            # extract current year
+            outcube = extract_time(cube, time.year, 1, 1, time.year, 12, 31)
+            # adjust time bounds to (year-01-01 00:00, year+1-01-01 00:00)
+            out_timecoord = outcube.coord("time")
+            start_date = datetime.datetime(time.year, 1, 1)
+            end_date = datetime.datetime(time.year + 1, 1, 1)
+            out_timecoord.bounds = np.array(
+                [
+                    out_timecoord.units.date2num(start_date),
+                    out_timecoord.units.date2num(end_date),
+                ]
+            )
+            # write output to file
+            logger.debug("Saving cube\n%s", outcube)
+            logger.debug("Setting time dimension to UNLIMITED while saving!")
+            save_variable(
+                outcube,
+                var["short_name"],
+                out_dir,
+                attributes,
+                unlimited_dimensions=["time"],
+            )
+
+    logger.info("Finished CMORizing %s", ", ".join(in_files))
+
+
+def _fix_coordinates(cube, definition):
+    """Fix coordinates."""
+    axis2def = {"T": "time", "X": "longitude", "Y": "latitude"}
+    axes = ["T", "X", "Y"]
+
+    for axis in axes:
+        coord_def = definition.coordinates.get(axis2def[axis])
+        if coord_def:
+            coord = cube.coord(axis=axis)
+            if axis == "T":
+                coord.convert_units("days since 1850-1-1 00:00:00.0")
+                coord.points = coord.core_points().astype("float64")
+                if coord.bounds is not None:
+                    coord.bounds = None
+
+            if len(coord.points) > 1:
+                if coord.bounds is not None:
+                    coord.bounds = None
+                coord.guess_bounds()
+            coord.standard_name = coord_def.standard_name
+            coord.var_name = coord_def.out_name
+            coord.long_name = coord_def.long_name
+
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize data."""
+    glob_attrs = cfg["attributes"]
+
+    logger.info(
+        "Starting cmorization for tier%s OBS files: %s",
+        glob_attrs["tier"],
+        glob_attrs["dataset_id"],
+    )
+    logger.info("Input data from: %s", in_dir)
+    logger.info("Output will be written to: %s", out_dir)
+    logger.info("CMORizing ESACCI-BIOMASS version %s", glob_attrs["version"])
+
+    for short_name, var in cfg["variables"].items():
+        filepattern = os.path.join(in_dir, var["filename"])
+        in_files = glob.glob(filepattern)
+        if "short_name" not in var:
+            var["short_name"] = short_name
+        if not in_files:
+            logger.info("%d: no data not found for variable %s", short_name)
+        else:
+            _extract_variable(in_files, var, cfg, out_dir)
```

Comment:
```suggestion
        if not in_files:
            msg = f"no data not found for variable {short_name}"
            raise ValueError(msg)
        _extract_variable(in_files, var, cfg, out_dir)
```

You probably want the code to stop here if no input files are found. If not, I would at least log an error (`logger.error`).